packages/@expo/cli/src/lint/lintAsync.ts

```diff
@@ -0,0 +1,115 @@
+import JsonFile from '@expo/json-file';
+import * as PackageManager from '@expo/package-manager';
+import spawnAsync from '@expo/spawn-async';
+import fs from 'fs/promises';
+import path from 'path';
+
+import { Log } from '../log';
+import { selectAsync } from '../utils/prompts';
+
+const WITH_PRETTIER = `module.exports = {
+  root: true,
+  extends: ["universe/native", "prettier"],
+  plugins: ["prettier", "expo"],
+  rules: {
+    "prettier/prettier": ["warn"],
+    "expo/no-env-var-destructuring": ["error"],
+    "expo/no-dynamic-env-var": ["error"],
+  },
+};
+
+`;
+
+const ESLINT_ONLY = `module.exports = {
+  root: true,
+  extends: ["universe/native"],
+  plugins: ["expo"],
+  rules: {
+    "expo/no-env-var-destructuring": ["error"],
+    "expo/no-dynamic-env-var": ["error"],
+  },
+};
+`;
+
+const setupLinting = async (projectRoot: string) => {
+  const result = await selectAsync(
+    'No eslint config found. Would you like to set up linting for this project?',
+    [
+      {
+        title: 'Yes, eslint only',
+        value: 'eslint',
+      },
+      {
+        title: 'Yes, eslint and prettier',
+        value: 'eslint-and-prettier',
+      },
+      {
+        title: 'No',
+        value: 'no',
+      },
+    ]
+  );
+
+  if (result === 'no') {
+    return;
+  }
+
+  const commandSegments = [
+    'expo',
+    'install',
+    'eslint',
+    'eslint-plugin-expo',
+    'eslint-config-universe', // TODO(Kadi): eslit-config-expo once published
+  ];
+
+  if (result === 'eslint-and-prettier') {
+    commandSegments.push('prettier');
+    commandSegments.push('eslint-config-prettier');
+    commandSegments.push('eslint-plugin-prettier');
+  }
+
+  // TODO(Kadi): how to add this as dev dependencies?
+  await spawnAsync('npx', commandSegments, {
+    stdio: 'inherit',
+    cwd: projectRoot,
+    env: { ...process.env },
+  });
+
+  await fs.writeFile(
+    path.join(projectRoot, '.eslintrc.js'),
+    result === 'eslint' ? ESLINT_ONLY : WITH_PRETTIER,
+    'utf8'
+  );
+
+  const scripts = JsonFile.read(path.join(projectRoot, 'package.json')).scripts;
+
+  await JsonFile.setAsync(
+    path.join(projectRoot, 'package.json'),
+    'scripts',
+    typeof scripts === 'object' ? { ...scripts, lint: 'eslint .' } : { lint: 'eslint .' },
+    { json5: false }
+  );
```

Comment:
Might be nice to not-overwrite existing linting scripts here. If users did configure this somehow, we can just tell them we found an existing "lint" script + ensure it's running `eslint`.

```suggestion
  if (scripts?.lint) {
    // log, skipped adding lint script, already exists:
    //   "lint": "eslint ." (the script we wanted to add)
  } else {
    await JsonFile.setAsync(
      path.join(projectRoot, 'package.json'),
      'scripts',
      typeof scripts === 'object' ? { ...scripts, lint: 'eslint .' } : { lint: 'eslint .' },
      { json5: false }
    );
  }
```

Note, something that I noticed on my end was that I need to explicitly add `eslint . --ext js,ts,tsx` to lint typescript files. Not sure if it's required here too though.

Reply:
It depends on the eslint config I think. E.g. with universe it doesn't need to be added.